consensus/polybft/blockchain_wrapper.go

```diff
@@ -0,0 +1,191 @@
+	"github.com/0xPolygon/polygon-edge/consensus"
+	"github.com/0xPolygon/polygon-edge/state"
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/umbracle/ethgo"
+	"github.com/umbracle/ethgo/contract"
+)
+
+// // ethereumBackend is an interface that wraps the methods called on ethereum backend protocol
+// type ethereumBackend interface {
+// 	// PeersLen returns the number of peers the node is connected to
+// 	PeersLen() int
+// 	// TxPool returns the current transaction pool
+// 	TxPool() *core.TxPool
+// 	// Broadcast block broadcasts the newly inserted block to the rest of the peers
+// 	BroadcastBlock(block *types.Block, propagate bool)
+// 	// GenesisGasLimit returns the initial gas limit for a block
+// 	GenesisGasLimit() uint64
+// }
+
+// blockchain is an interface that wraps the methods called on blockchain
+type blockchainBackend interface {
+	// CurrentHeader returns the header of blockchain block head
+	CurrentHeader() *types.Header
+
+	// CommitBlock commits a block to the chain.
+	CommitBlock(stateBlock *StateBlock) error
+
+	// NewBlockBuilder is a factory method that returns a block builder on top of 'parent'.
+	NewBlockBuilder(parent *types.Header, coinbase types.Address) (blockBuilder, error)
+
+	// ProcessBlock builds a final block from given 'block' on top of 'parent'.
+	ProcessBlock(parent *types.Header, block *types.Block) (*StateBlock, error)
+
+	// GetStateProviderForBlock returns a reference to make queries to the state at 'block'.
+	GetStateProviderForBlock(block *types.Header) (contract.Provider, error)
+
+	// GetStateProvider returns a reference to make queries to the provided state.
+	GetStateProvider(transition *state.Transition) contract.Provider
+
+	// GetHeaderByNumber returns a reference to block header for the given block number.
+	GetHeaderByNumber(number uint64) (*types.Header, bool)
+
+	// GetHeaderByHash returns a reference to block header for the given block hash
+	GetHeaderByHash(hash types.Hash) (*types.Header, bool)
+
+	// GetSystemState creates a new instance of SystemState interface
+	GetSystemState(config *PolyBFTConfig, provider contract.Provider) SystemState
+
+	SubscribeEvents() blockchain.Subscription
+}
+
+var _ blockchainBackend = &blockchainWrapper{}
+
+type blockchainWrapper struct {
+	executor   *state.Executor
+	blockchain *blockchain.Blockchain
+}
+
+// CurrentHeader returns the header of blockchain block head
+func (p *blockchainWrapper) CurrentHeader() *types.Header {
+	return p.blockchain.Header()
+}
+
+// CommitBlock commits a block to the chain
+func (p *blockchainWrapper) CommitBlock(stateBlock *StateBlock) error {
+	// logs := buildLogsFromReceipts(stateBlock.Receipts, stateBlock.Block.GetHeader())
+	// status, err := p.blockchain.WriteBlockAndSetHead(stateBlock.Block,
+	// stateBlock.Receipts, logs, stateBlock.State, true)
+	// if err != nil {
+	// 	return err
+	// } else if status != core.CanonStatTy {
+	// 	return fmt.Errorf("non canonical change")
+	// }
```

Comment:
Leftover?

Reply:
Removed in another pr https://github.com/0xPolygon/polygon-edge/pull/789